packages/twenty-front/src/modules/object-record/types/BaseObjectRecord.ts

```diff
@@ -0,0 +1,4 @@
+export type BaseObjectRecord = {
```

Comment:
Base seems a bit too vague and I'm not sure that's really what we should consider as the base of all object records, because some of them might not have __typename.

Could BaseCacheObjectRecord help convey the fact that we're dealing with a type that will be used for cache related stuff ?

Reply:
This is just an extract of the original type but I'm wondering if we shouldn't use this as an opportunity to be more precise.

Reply:
I was thinking of this type agnostically of caching
What I'm searching here is having a type I can infer static fields common to any object record
Have been considering:
- CommonObjectRecord
- DeterministicObjectRecord
Etc

In the best of the world would be added common fields such as deletedAt etc
That would easy typing through the app

Charles suggested BaseObjectRecord to keep consistency with twenty-server

Reply:
That should be discussed first because we need to first make sure that we assume that some props are present on all object records, and this should be enforced in the backend and maybe and/or with zod at least because that could change for a specific object and the runtime wouldn't reflect our typing, thus creating nasty bugs not caught at compile time.